### PR TITLE
cleaned up dockerfiles and added jupyter build

### PR DESCRIPTION
**Pull request by epinzur**
removed unused `Dockerfile.notebook`

trying to see if we can use `Dockerfile.release` in place of `Dockerfile.integration` since they are so similar.

Added new `Dockerfile.jupyter` to create a mult-arch release of our system based off a Jupyter notebook environment with pre-installed kaskada services and clients.

### CODE REVIEW

File: Dockerfile.jupyter [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# Install the pulsar python client and the kaskada python client
+RUN pip install pulsar-client kaskada --no-cache-dir
+
+RUN mkdir -p /home/jovyan/.cache/kaskada/bin
+COPY --chmod=755 ./release/${TARGETOS}/${TARGETARCH}/cli            /home/jovyan/.cache/kaskada/bin/kaskada-cli
+COPY --chmod=755 ./release/${TARGETOS}/${TARGETARCH}/wren           /home/jovyan/.cache/kaskada/bin/kaskada-manager
+COPY --chmod=755 ./release/${TARGETOS}/${TARGETARCH}/sparrow-main   /home/jovyan/.cache/kaskada/bin/kaskada-engine
+
+ENV KASKADA_DISABLE_DOWNLOAD=true
```
Comment (epinzur) [inline Dockerfile.jupyter:31]: should look very similar to `Dockerfile.release`

File: Dockerfile.jupyter.dockerignore [hunk 2]
```diff
@@ -0,0 +1,7 @@
+*
+!release/linux/amd64/sparrow-main
+!release/linux/amd64/wren
+!release/linux/amd64/cli
+!release/linux/arm64/sparrow-main
+!release/linux/arm64/wren
+!release/linux/arm64/cli
```
Comment (epinzur) [inline Dockerfile.jupyter.dockerignore:7]: should look very similar to `Dockerfile.release.dockerignore`

**Comment by epinzur** [comment]
closes #342

File: .github/workflows/ci_engine.yml [hunk 3]
```diff
@@ -66,14 +66,14 @@ jobs:
...
-          path: ci-bin/wren/
+          name: manager-binary
+          path: release/linux/amd64/
 
       - name: Get Sparrow binary
         uses: actions/download-artifact@v3
         with:
           name: engine-debug
-          path: ci-bin/engine/
+          path: release/linux/amd64/
```
Comment (epinzur) [inline .github/workflows/ci_engine.yml:76]: i realize it is a bit odd copying a debug build to a release path, but this works.  Maybe in the future we can drop the `release` copy path in `Dockerfile.release`

File: .github/workflows/release_engine.yml [hunk 4]
```diff
@@ -289,6 +290,24 @@ jobs:
             org.opencontainers.image.revision=${{ github.sha }}
             org.opencontainers.image.licenses=Apache-2.0
 
+      - name: Build and push Docker images for Jupyter
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          platforms: linux/amd64, linux/arm64
+          file: Dockerfile.release
```
[the remark this answers is not in the document]
Comment (epinzur) [reply .github/workflows/release_engine.yml]: yes, good catch!

**Review by therapon — APPROVED** [review]

**Pull request merged**